portal-kernel/src/com/liferay/portal/kernel/messaging/DestinationFactoryUtil.java

```diff
@@ -39,6 +39,6 @@ private DestinationFactoryUtil() {
 	private static volatile DestinationFactory _destinationFactory =
 		ServiceProxyFactory.newServiceTrackedInstance(
 			DestinationFactory.class, DestinationFactoryUtil.class,
-			"_destinationFactory", true);
+			"_destinationFactory", false, true);
```

Comment:
@JanisZhang The commit message is not a good reason for this change. Basically, the original logic in https://github.com/liferay-core-infra/liferay-portal/commit/1a06e53b34e64c6ad5b9fc1c37904982dfb15197 does wait for the target service in the actual get method instead of in the proxy. That's why they were replaced with blocking usages of `ServiceProxyFactory`.

For `DestinationFactoryUtil`, I think it's easier to change. It has only 1 actual usage in portal-impl where a service latch is used to wait for `DestinationFactory`. All other usages are from integration tests where you can just `@Inject`.

For `MessageBusUtil`, please double check and send separately. Try to eliminate the usages in modules first, and then review the usages in core.